batch/test/test_batch.py

```diff
@@ -566,3 +566,18 @@ def test_verify_no_access_to_metadata_server(self):
         status = j.wait()
         assert status['state'] == 'Failed', status
         assert "Connection timed out" in j.log()['main'], (j.log()['main'], status)
+
+    def test_user_authentication_within_job(self):
+        batch = self.client.create_batch()
+        cmd = ['bash', '-c', f'hailctl batch list']
```

Comment:
This is not the best test because it's going to list all possible batches that have been created ever (I think). I tried to add a CLI for billing projects in batch. That would be better to use. See #9385

The other option is to see if we can add paging to the CLI interface. But that seems like more work. I'm not sure why my PR keeps failing test_query.


Actually, there's a `-q` parameter. Maybe we can use that to select only running batches. That would probably work :)

Reply:
cool, I'll take a look!

I think that for most of the tests that get run e.g. inside PRs, we'll always have a more-or-less limited number of batches (~120?) but it would be definitely good to restrict this. I can also probably do `hailctl auth list` and check for the expected namespace, since the purpose is just to test that the auth is working correctly; do you think that would work?

Reply:
Yes. That should work too. I'm worried about the deploy tests.

Reply:
I think figuring out the `-q running` would be fine as well.